python/ray/tune/examples/xgboost_example.py

```diff
@@ -1,47 +1,74 @@
+from typing import Dict, List
 import sklearn.datasets
 import sklearn.metrics
 import os
+import numpy as np
 from ray.tune.schedulers import ASHAScheduler
 from sklearn.model_selection import train_test_split
 import xgboost as xgb
 
 from ray import tune
-from ray.tune.integration.xgboost import TuneReportCheckpointCallback
+from ray.tune.integration.xgboost import (
+    TuneReportCheckpointCallback,
+    TuneReportCallback,
+)
 
 
 def train_breast_cancer(config: dict):
     # This is a simple training function to be passed into Tune
     # Load dataset
     data, labels = sklearn.datasets.load_breast_cancer(return_X_y=True)
-    # Split into train and test set
-    train_x, test_x, train_y, test_y = train_test_split(data, labels, test_size=0.25)
-    # Build input matrices for XGBoost
-    train_set = xgb.DMatrix(train_x, label=train_y)
-    test_set = xgb.DMatrix(test_x, label=test_y)
-    # Train the classifier, using the Tune callback
-    xgb.train(
-        config,
-        train_set,
-        evals=[(test_set, "eval")],
-        verbose_eval=False,
-        callbacks=[TuneReportCheckpointCallback(filename="model.xgb")],
-    )
+
+    if config.pop("use_cv", False):
+        # For CV, we need to average over a list of results form folds
+        def average_cv_folds(results_dict: Dict[str, List[float]]) -> Dict[str, float]:
+            return {k: np.mean(v) for k, v in results_dict.items()}
+
+        train_set = xgb.DMatrix(data, label=labels)
+
+        # Run CV, using the Tune callback
+        xgb.cv(
+            config,
+            train_set,
+            verbose_eval=False,
+            stratified=True,
+            # Checkpointing is not supported for CV
+            callbacks=[TuneReportCallback(results_postprocessing_fn=average_cv_folds)],
+        )
```

Comment:
Instead of using these clauses, can we just define another highlevel function (`train_breast_cancer_cv`) and use that? The point of these examples is to show how simple training functions can be defined, but including all in one makes it hard to digest.

Reply:
Good idea, done